### PR TITLE
React: improved types quality for CSS properties

### PR DESCRIPTION
**Pull request by ErikCupal**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.mozilla.org/en-US/docs/Web/CSS/Reference
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

### CODE REVIEW

**Comment by ErikCupal** [comment]
Hi guys 😃! I improved the types for CSS properties so that we get a little better typechecking. I followed the CSS reference on MDN - I followed the Formal Syntax for every prop if there was one. I used `string` type or `string` | `number` type when it was not possible to create more narrowed types (hopefully sometime [in the future](https://github.com/Microsoft/TypeScript/issues/6579)).

Hope you like it. Comment if you feel there could be an improvement.

**Comment by ErikCupal** [comment]
Ahh, I tested just the package itself, not the others...

**Comment by ErikCupal** [comment]
There are few errors in tests of other libs (aphrodite, victory) - it seems they are actually mistakes in the tests. Look [here](https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/227330152?utm_source=github_status&utm_medium=notification).

**Comment by mhegazy** [comment]
@ErikCupal please fix the other ones, we can not merge the PR as long as the CI server is failing.

File: types/react/index.d.ts [hunk 1]
```diff
@@ -665,52 +665,52 @@ declare namespace React {
...
          * Defines a length of time to elapse before an animation starts, allowing an animation to begin execution some time after it is applied.
          */
-        animationDelay?: CSSWideKeyword | any;
+        animationDelay?: CSSWideKeyword | string | number;
 
         /**
          * Defines whether an animation should run in reverse on some or all cycles.
          */
-        animationDirection?: CSSWideKeyword | any;
+        animationDirection?: CSSWideKeyword | "normal" | "reverse" | "alternate" | " alternate-reverse";
```
Comment (vsiao) [inline types/react/index.d.ts]: extra space

File: types/react/index.d.ts [hunk 2]
```diff
@@ -2657,7 +2692,7 @@ declare namespace React {
         isRequired: Validator<T>;
     }
 
-    type ValidationMap<T> = { [K in keyof T]?: Validator<T> };
```
Comment (vsiao) [inline types/react/index.d.ts]: unintentional change?

File: types/react/index.d.ts [hunk 3]
```diff
@@ -1184,26 +1219,26 @@ declare namespace React {
...
+        fontKerning?: CSSWideKeyword | "auto" | "normal" | "none";
 
         /**
          * Specifies the size of the font. Used to compute em and ex units.
          * See CSS 3 font-size property https://www.w3.org/TR/css-fonts-3/#propdef-font-size
          */
         fontSize?: CSSWideKeyword |
-                   "xx-small" | "x-small" | "small" | "medium" | "large" | "x-large" | "xx-large" |
-                   "larger" | "smaller" |
-                   CSSLength | CSSPercentage;
```
Comment (vsiao) [inline types/react/index.d.ts]: did you mean to remove this indent? can you keep it indented (or move to 4 spaces), please?

**Review by vsiao — COMMENTED** [review]
Thanks! this seems valuable. small nit: I'd prefer if some of these declarations were split over multiple lines for better readability.

**Comment by andy-ms** [comment]
Please see the review comments.

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has failing CI tests and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Comment by ErikCupal** [comment]
Thanks a lot for comments :). Sorry, I've been busy lately and completely forgot about this PR. I'll take a look at it when I have time again.

**Pull request closed without merging**